Source/monster.cpp

```diff
@@ -4574,18 +4561,13 @@ void SyncMonsterAnim(Monster &monster)
 	case MonsterMode::Charge:
 		graphic = MonsterGraphic::Attack;
 		monster.AnimInfo.CurrentFrame = 1;
-		monster.AnimInfo.NumberOfFrames = monster.MType->GetAnimData(MonsterGraphic::Attack).Frames;
 		break;
 	default:
 		monster.AnimInfo.CurrentFrame = 1;
-		monster.AnimInfo.NumberOfFrames = monster.MType->GetAnimData(MonsterGraphic::Stand).Frames;
 		break;
 	}
 
-	if (monster.MType->GetAnimData(graphic).CelSpritesForDirections[mdir])
-		monster.AnimInfo.pCelSprite = &*monster.MType->GetAnimData(graphic).CelSpritesForDirections[mdir];
-	else
-		monster.AnimInfo.pCelSprite = nullptr;

```

Comment:
Attempting to render a monster with a `nullptr` for pCelSprite is logged as an error, but I hadn't seen that actually fire in my testing. I can restore this code in `ChangeAnimationData` if it's actually hit on some levels.

Reply:
If we change `ChangeAnimationData` to also set `pCelSprite` to `nullptr`, it wouldn't be any difference (see comment in `ChangeAnimationData).
The only case this could happen in normal game that comes to my mind is that golems don't have a Stand graphics. So they could end up with `nullptr`, but I'm not sure if this is the case.
If you want you can check it again. 🙂